lib/class-wp-block-supports.php

```diff
@@ -49,6 +56,30 @@ public static function init() {
 		$instance->register_attributes();
 	}
 
+	/**
+	 * Callback hooked to the register_block_type_args filter.
+	 *
+	 * This hooks into block registration to wrap the render_callback
+	 * of dynamic blocks with a closure that keeps track of the
+	 * current block to be rendered.
+	 *
+	 * @param array $args Block attributes.
+	 * @return array Block attributes.
+	 */
+	public static function track_block_to_render( $args ) {
```

Comment:
I'd leave this one out of the class because it's just a "compat" thing while we fix the root issue on Core.

Reply:
done at https://github.com/WordPress/gutenberg/pull/26356/commits/1a0e049a93be8d12f70c0a7bca01f26392757515